docs/development/pigs/pig-007.rst

```diff
@@ -0,0 +1,409 @@
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+
+The current class names are given in parentheses. Optional model classes, where
+the need is unclear are marked with a question mark.
+
+Parametric models are named after their parametrization, such as `Powerlaw` or
+`Gaussian`. In case the name is not unique (e.g. for a gaussian spectral and spatial
+model, the corresponding prefix should be used, such as `SpectralGaussian` or
+`SpatialGaussian` or `SpatialConstant` and `SpectralConstant`).
+
+In addition there is the `SourceModels`  class to represent a sum of `SourceModel`
+objects.
+
+
+Introduction of background models
+---------------------------------
+For any kind of IACTs analysis hadronic background models are required, which model
+the residual hadronic (gamma-like) background emission from templates. They differ
+from source templates, because they already have the IRFs applied. We propose
+the introduction of the following two model classes:
+
+`BackgroundModel`
+~~~~~~~~~~~~~~~~~
+Base class for all background models.
+
+`BackgroundIRFModel`
+~~~~~~~~~~~~~~~~~~~~
+Class to represent a template background model. It is initialized with a
+background map template and introduces additional background parameters to
+adjust the spectral shape and amplitude. This model can only be evaluated on a
+fixed grid, defined by the input map. The model is interpolated in the preparation
+step. No IRFs are applied.
+
+.. code-block::
+
+    from gammapy.cube import BackgroundTemplate
+
+    background_map = Map()
+    background = BackgroundTemplate(template=background_map, norm=1, tilt=0.1)
+    npred = background.npred
+
+
+
+Introduction of "forward folded" models
+---------------------------------------
+To combine parametric model components with their correspoding instrument response
+functions, such as PSF, energy dispersion and exposure, we propose the introduction
+of "forward folded" models. These take a fixed data grid, typically defined by the
+exposure map or effective area table, on which the model is evaluated. After application of
+instrument reponse functions, the number of prediced counts is returned. This can be
+either accessed by a `.npred` property or is returned by `.evaluate()`. We propose
+to introduce the following classes:
+
+`SpectralIRFModel`
+~~~~~~~~~~~~~~~~~~
+A "forward folded" spectral model, that applies energy dispersion and effective
+area to a `SpectralModel`. It corresponds in functionality to the current `CountsPredictor`.
+
+.. code-block::
+
+    from gammapy.spectrum import Powerlaw, SpectralIRFModel
+
+    pwl = Powerlaw()
+    spectral_irf_model = SpectralIRFModel(model=pwl, aeff=aeff, livetime=livetime,
+                                          edisp=edisp)
+    npred = spectral_irf_model.npred
+    dnde = spectral_irf_model.dnde
```

Comment:
@registerrier I agree there is no strong need to expose this in the API. It was mainly inspired from the current `MapEvaluator.compute_dnde()` and `MapEvaluator.compute_flux()` methods. However it might be still useful for debugging...